packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+    const prefix = 'lg-passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const descriptionId = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const labelId = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
+    });
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const { value, handleChange } = useControlledValue(valueProp, onChangeProp);
+
+    if (!label && !ariaLabelledbyProp && !rest['aria-label']) {
+      console.warn(
+        'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided',
+      );
+    }
+
+    // If disabled then hide password
+    useEffect(() => {
+      if (disabled) setShowPassword(false);
+    }, [disabled]);
+
+    const handleTogglePasswordClick = () => setShowPassword(s => !s);
+
+    /**
+     * The overall state of the component
+     */
+    const state: States = Array.isArray(stateNotifications)
+      ? getStateFromArray(stateNotifications)
+      : stateNotifications;
+
+    /**
+     * Checks if there are any notifications
+     */
+    const hasNotifications: boolean =
+      !ariaDescribedbyProp && Array.isArray(stateNotifications);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <div className={className} ref={forwardedRef}>
+          {label && (
+            <Label
+              id={labelId}
+              className={cx(labelBaseStyles, {
+                [labelLargeOverrideStyles]: sizeVariant === SizeVariant.Large,
+              })}
+              htmlFor={inputId}
+              disabled={disabled}
+            >
+              {label}
+            </Label>
+          )}
+          <div className={inputWrapperStyles}>
+            <input
+              type={showPassword ? 'text' : 'password'}
+              value={value}
+              id={inputId}
+              autoComplete={autoComplete}
+              aria-describedby={descriptionId}
+              aria-labelledby={labelId}
+              aria-disabled={disabled}
+              aria-invalid={state === States.Error || state === States.Warning}
+              className={cx(
+                inputBaseStyles,
+                inputSizeStyles[sizeVariant],
+                inputBaseThemeStyles[theme],
+                {
+                  [inputThemeStyles[theme][state]]: !disabled,
+                  [cx(
+                    inputDisabledBaseStyles,
+                    inputDisabledThemeStyles[theme],
+                  )]: disabled,
+                  [inputIconSizeStyles[sizeVariant]]:
+                    !hasNotifications && state !== States.None,
+                },
+              )}
+              onChange={handleChange}
+              readOnly={disabled ? true : false}
+              {...rest}
+            />
+            {/* Visual icons inside the input will only render if aria-describedby is set and the state is not `none`. `none` does not need a visible icon */}
```

Comment:
This comment is misleading

Reply:
hows this: If a custom message container is used, an icon will render inside the input to represent the state of the input. In the case that stateNotification === `none`, no icon will appear.